blog/posts/announcing-melange-5.md

```diff
@@ -0,0 +1,285 @@
+twitter: '@_anmonteiro'
+---
+
+We are excited to announce the release of Melange 5, the compiler for OCaml
+that targets JavaScript.
+
+A lot of goodies went into this release! While our focus was mostly on features
+that make it easy to express more JavaScript constructs and supporting OCaml
+5.3, we also managed to fit additional improvements in the release: better
+editor support for Melange `external`s, code generation improvements, and
+better compiler output for generated JS. The most notable feature we're
+shipping in Melange 5 is support for JavaScript's [dynamic
+`import()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import),
+which we'll describe in detail below.
+
+Read on for the highlights.
+
+## Dynamic `import()` without sacrificing type safety
+
+Support for JavaScript's dynamic `import()` is probably what I'm most excited
+about in this Melange release. In Melange 5, we're releasing support for
+JavaScript's `import()` via a new function in `melange.js`, `Js.import: 'a ->
+'a promise`. I gave a small preview of dynamic `import()` during my [Melange
+talk](https://www.youtube.com/watch?v=3oCXT-ycHHs) at [Fun OCaml
+2024](https://fun-ocaml.com).
+
+`Js.import` is **type-safe** and **build system-compatible**. Let's break it
+down:
+
+1. **build system-compatible**: dynamic `import()`s in Melange work with Dune
+   out of the box: as usual, you must specify your `(library ..)` dependencies
+   in the `dune` file. At compile time, Melange will be aware of the
+   dynamically imported module locations to emit the arguments to
+   `import("/path/to/module.js)` automatically.
+2. **type-safe**: your OCaml code is still aware of its dependencies at
+   compile-time, but Melange will skip emitting static JavaScript `import ..`
+   declarations if the values are exclusively used through `Js.import(..)`.
+
+### Dynamically `import()`ing OCaml code
+
+The example below makes it clear: we import the entire `Stdlib.Int` module,
+specify its type signature, and observe that no static `import`s appear in the
+resulting JavaScript:
+
+```js
+// dynamic_import_int.re
+
+module type int = (module type of Int);
+
+let _: Js. Promise.t(unit) = {
+  Js.import((module Stdlib.Int): (module int))
+  |> Js.Promise.then_((module Int: int) =>
+      Js.Promise.resolve(Js.log(Int.max))
+  );
+};
+```
+
+```js
+// dynamic_import_int.js
+
+import("melange/int.js").then(function (Int) {
+  return Promise.resolve((console.log(Int.max), undefined));
+});
+```
+
+### Dynamically `import()`ing JavaScript from OCaml
+
+One of Melange's primary operating principles is the ability to support
+seamless interop with JavaScript constructs. As such, we implemented `import()`
+in a way that also allows importing JS modules dynamically: you can call
+`Js.import` on JavaScript values declared with `external`. The abstractions
+compose nicely here to produce the expected result. Check out this example of
```

Comment:
```suggestion
compose nicely to produce the expected result. Check out this example of
```

Otherwise sounds like they don't compose nicely somewhere else 😄